SubMotionVR Mod/SubMotion/SubMotion/Patchers/ArmsController.cs

```diff
@@ -9,38 +9,87 @@
+        }
+
+        protected void Initialize(ArmsController _armsController)
+        {
+            this.armsController = _armsController;
+            Transform parent = this.armsController.player.camRoot.transform;
+
+            this.rightController = new GameObject("rightController");
+            this.rightController.transform.parent = parent;
+
+            this.leftController = new GameObject("leftController");
+            this.leftController.transform.parent = parent;
+        }
+
+        public void UpdateHandPositions() {
+            InventoryItem heldItem = Inventory.main.quickSlots.heldItem;
+
+            this.rightController.transform.localPosition = InputTracking.GetLocalPosition(VRNode.RightHand) + new Vector3(0f, -0.02f, -0.2f);
+            this.rightController.transform.localRotation = InputTracking.GetLocalRotation(VRNode.RightHand) * Quaternion.Euler(35f, 190f, 270f);
+
+            leftController.transform.localPosition = InputTracking.GetLocalPosition(VRNode.LeftHand) + new Vector3(0f, 0.02f, -0.2f);
+            leftController.transform.localRotation = InputTracking.GetLocalRotation(VRNode.LeftHand) * Quaternion.Euler(270f, 90f, 0f);
+
+            if (heldItem.item.GetComponent<PropulsionCannon>()) {
+                __instance.ik.solver.leftHandEffector.target = null;
+                __instance.ik.solver.rightHandEffector.target = null;
+            } else if (heldItem.item.GetComponent<StasisRifle>()) {
+                __instance.ik.solver.leftHandEffector.target = null;
+                __instance.ik.solver.rightHandEffector.target = null;
+            } else {
+                __instance.ik.solver.leftHandEffector.target = leftController.transform;
+                __instance.ik.solver.rightHandEffector.target = rightController.transform;
+            }
+        }
+    }
+
+
+    [HarmonyPatch(typeof(ArmsController))]
+    [HarmonyPatch("Start")]
+    internal class ArmsController_Start_Patch
+    {
+        [HarmonyPostfix]
+        public void PostFix(ArmsController __instance)
+        {
+            if (!VRSettings.enabled) {
+                return;
+            }
+
+            VRHandsController.instance.Initialize(__instance);
+        }
+    }
+
+    [HarmonyPatch(typeof(ArmsController))]
     [HarmonyPatch("Update")]
     internal class ArmsController_Update_Patch
     {
 
         [HarmonyPostfix]
         public void Postfix(ArmsController __instance, FullBodyBipedChain ik, PlayerTool tool)
         {
-            if ((VRSettings.enabled && Player.main.motorMode != Player.MotorMode.Vehicle && !__instance.player.cinematicModeActive) || __instance.pda.isActiveAndEnabled)
+            if (!VRSettings.enabled) {
+                return;
+            }
+
+            if ((Player.main.motorMode != Player.MotorMode.Vehicle && !__instance.player.cinematicModeActive) || __instance.pda.isActiveAndEnabled)
             {
-                GameObject rightController = new GameObject("rightController");

```

Comment:
yeah I understand what you are saying, and the concept behind it, by creating a new object every frame. What I don't know is after that object is "created and moved to pos/rot" is that that.. and does it create a new object for net frame to move the hand into a new position, and doing this every frame is that why there is tracking? Cause I tried to move all this around to where the objects were created during the Start, and update only had the tracking. Nothing would track though, tried a few different things and couldn't get there to be any tracking anywhere else unless I had the objects in the update function with the rest of the code. I will try again, but that aside, have had no issues how it is now and no crashes. I can play for a long time and test to find out if it will/does crash. I understand though what you are seeing/talking about and what could be. As of right now though, everything works awesome in game as you are playing.